packages/pluggableWidgets/accordion-web/src/components/Accordion.tsx

```diff
@@ -0,0 +1,42 @@
+import { createContext, createElement, Dispatch, ReactElement, useMemo, useReducer, useRef } from "react";
+
+import AccordionGroup, { AccGroup } from "./AccordionGroup";
+import { AccordionGroupsReducerAction, getAccordionGroupsReducer } from "../utils/AccordionGroupStateReducer";
+import { AccordionContainerProps } from "../../typings/AccordionProps";
+
+export interface AccordionProps extends Pick<AccordionContainerProps, "class" | "style" | "tabIndex"> {
+    id: string;
+    groups: AccGroup[];
+    collapsible: boolean;
+    singleExpandedGroup?: boolean;
+}
+
+export const AccordionGroupsDispatch = createContext<Dispatch<AccordionGroupsReducerAction> | undefined>(undefined);
+
+export default function Accordion(props: AccordionProps): ReactElement | null {
+    const { id, class: classNames, style, tabIndex, groups, collapsible, singleExpandedGroup } = props;
+
+    const previousGroupsPropValue = useRef(groups);
+
+    const [accordionGroups, accordionGroupsDispatch] = useReducer(
+        getAccordionGroupsReducer(collapsible ? (singleExpandedGroup ? "single" : "multiple") : "all"), // the accordion group reducer function doesn't need to change during the lifetime of this component, since the singleExpandedGroup won't change.
+        groups.map(group => (collapsible ? { ...group, collapsed: true } : { ...group, collapsed: false }))
+    );
+
+    if (groups !== previousGroupsPropValue.current) {
```

Comment:
Dont we need to compare using deep-equal here?

Reply:
No the array will always be different when there is a change in one of the group values, because of the `useMemo` implementation in `src/Accordion.tsx`